osquery/tables/system/linux/startup_items.cpp

```diff
@@ -81,6 +83,113 @@ void genAutoStartScripts(const std::string& sysdir, QueryData& results) {
   }
 }
 
+int bus_iter_get_basic_and_next(DBusMessageIter* iter,
+                                int type,
+                                void* data,
+                                bool next) {
+  if (dbus_message_iter_get_arg_type(iter) != type) {
+    return -1;
+  }
+
+  dbus_message_iter_get_basic(iter, data);
+
+  if (dbus_message_iter_next(iter) != next) {
+    return -1;
+  }
+
+  return 0;
+}
+
+void genSystemDItems(QueryData& results) {
+  DBusConnection* c;
+  DBusError error;
+  DBusMessage *systemd_message = nullptr, *reply = nullptr;
+  DBusMessageIter iter, sub, sub2;
+
+  dbus_error_init(&error);
+  c = dbus_bus_get(DBUS_BUS_SYSTEM, &error);
+
+  if (c == NULL || dbus_error_is_set(&error)) {
+    VLOG(1) << "Error dbus failed: " << error.name << error.message;
+    goto finish;
+  }
+
+  systemd_message =
+      dbus_message_new_method_call("org.freedesktop.systemd1",
+                                   "/org/freedesktop/systemd1",
+                                   "org.freedesktop.systemd1.Manager",
+                                   "ListUnits");
+  if (!systemd_message) {
+    VLOG(1) << "Error failed to create dbus message";
+    goto finish;
+  }
+  reply =
+      dbus_connection_send_with_reply_and_block(c, systemd_message, -1, &error);
+  if (dbus_error_is_set(&error)) {
+    VLOG(1) << "Error dbus failed: " << error.name << error.message;
+    goto finish;
+  }
+
+  if (!dbus_message_iter_init(reply, &iter) ||
+      dbus_message_iter_get_arg_type(&iter) != DBUS_TYPE_ARRAY ||
+      dbus_message_iter_get_element_type(&iter) != DBUS_TYPE_STRUCT) {
+    VLOG(1) << "Error failed to parse reply";
+    goto finish;
+  }
+  dbus_message_iter_recurse(&iter, &sub);
+
+  while (dbus_message_iter_get_arg_type(&sub) != DBUS_TYPE_INVALID) {
+    Row r;
+    const char *id, *description, *load_state, *active_state, *sub_state,
+        *following, *unit_path, *job_type, *job_path;
+    std::uint32_t job_id = 0U;
+    if (dbus_message_iter_get_arg_type(&sub) != DBUS_TYPE_STRUCT) {
+      VLOG(1) << "Error failed to parse reply";
+      goto finish;
+    }
+    dbus_message_iter_recurse(&sub, &sub2);
+
+    if (bus_iter_get_basic_and_next(&sub2, DBUS_TYPE_STRING, &id, true) < 0 ||
```

Comment:
What is the goal of acquiring these values (description, load_state, sub_state, following, etc) if they are not used in the returned row data?

Reply:
I think they are being acquired to move the read pointer forward; I'm in the process of updating the PR and I'll see if there's a way to skip fields